keras_cv/models/segmentation/__internal__/segmentation_head.py

```diff
@@ -0,0 +1,109 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class SegmentationHead(layers.Layer):
+    """Prediction head for the segmentation model
+
+    The head will take the output from decoder (eg FPN or ASPP), and produce production
+    (pixel level classifications) as the output for the model.
+
+    Args:
+        classes: int, the number of output classes for the prediction.
+        convs: int, the number of conv2D layers that are stacked before the final
+            classification layer. Default to 2.
+        channels: int, the number of filter/channels for the the conv2D layers. Default
+            to 256.
+        activations: str or 'tf.keras.activations', activation functions between the
+            conv2D layers and the final classification layer. Default to 'relu'
+
+    Sample code
+    ```python
+    # Mimic a FPN output dict
+    p3 = tf.ones([2, 32, 32, 3])
+    p4 = tf.ones([2, 16, 16, 3])
+    p5 = tf.ones([2, 8, 8, 3])
+    inputs = {3: p3, 4: p4, 5: p5}
+
+    head = SegmentationHead(classes=11)
+
+    output = head(inputs)
+    # output tensor has shape [2, 32, 32, 11]. It has the same resolution as the p3.
+    ```
+    """
+
+    def __init__(self, classes, convs=2, channels=256, activations="relu", **kwargs):
+        """"""
+        super().__init__(**kwargs)
+        self.classes = classes
+        self.convs = convs
+        self.channels = channels
+        self.activations = activations
+
+        self._conv_layers = []
+        self._bn_layers = []
+        for i in range(self.convs):
+            conv_name = "segmentation_head_conv_{}".format(i)
+            self._conv_layers.append(
+                tf.keras.layers.Conv2D(
+                    name=conv_name,
+                    filters=self.channels,
+                    kernel_size=3,
+                    padding="same",
+                    use_bias=False,
+                )
+            )
+            norm_name = "segmentation_head_norm_{}".format(i)
+            self._bn_layers.append(tf.keras.layers.BatchNormalization(name=norm_name))
+
+        self._classification_layer = tf.keras.layers.Conv2D(
+            name="segmentation_output",
+            filters=self.classes,
+            kernel_size=1,
+            padding="same",
+        )
+
+    def call(self, inputs):
+        """Forward path for the segmentation head.
+
+        For now, it accepts the output from the decoder only, which is a dict with int
```

Comment:
specify the shape of decoder output here?

Reply:
Oh, it is included in the docstring above (as the code example.)